### PR TITLE
All claims 1dot4 mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
Adds separation / severance pay and training pay questions to the form, via one page that conditionally expands more questions based on qualifying answers.

## Testing done
tested locally, added some unit tests for the different form views and submit possibilities

## Testing Plan
Test this thing out on staging?

## Screenshots

Basic screen:
screen shot 2018-08-23 at 4 27 58 pm

-----

Selecting 'yes' to the separation pay question yields:
screen shot 2018-08-23 at 4 28 52 pm

-----

Selecting 'yes' on the training pay question yields:
screen shot 2018-08-23 at 4 28 31 pm

-----

## Acceptance Criteria (Definition of Done)
- [ ] Form page reviewed by content and UX
- [x] Page is able to be submitted with all required questions answered

#### Applies to all PRs

- [x] Appropriate logging
- [x] Documentation has been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

**Review by jbalboni — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/tests/pages/separationTrainingPay.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        }}
+        formData={{}}/>
+    );
+
+    expect(form.find('.form-radio-buttons').length).to.equal(2);
+    expect(form.find('input').length).to.equal(6); // 4 radios + year input + checkbox
+    expect(form.find('select').length).to.equal(3); // month, day, service branch
+  });
+
+  it('should submit when all required info filled', () => {
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/tests/pages/separationTrainingPay.unit.spec.jsx]: Is this test needed? Seems like it's mostly testing the forms library, which is tested separately.

The other tests in this file look good, though.

I think generally we want to make sure
1) All the fields we expect show up
2) We can submit with the minimum required fields
3) All the fields we expect to be required are

**Review by cvalarida — APPROVED** [review]
I said a bunch of stuff, but they were all pretty small. Looks good!

File: src/applications/disability-benefits/all-claims/config/schema.js [hunk 2]
```diff
@@ -1,5 +1,15 @@
 // This file to be moved to vets-json-schema repo before form hits production
 
+const serviceBranches = [
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/config/schema.js]: No, best I can tell this is a subset of all service branches.

File: src/applications/disability-benefits/all-claims/tests/pages/separationTrainingPay.unit.spec.jsx (hunk 1 above)
Comment (jbalboni) [reply src/applications/disability-benefits/all-claims/tests/pages/separationTrainingPay.unit.spec.jsx]: I think this test is useful in that it tests that the validation logic is what we expect and there's not some hidden issue that is blocking moving to the next page.

Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/tests/pages/separationTrainingPay.unit.spec.jsx]: Yeah, I guess the thinking with this test was that clicking certain options made _more_ stuff required, so I wanted to test the base case that the form would submit with all of the conditionally required stuff filled out. But yeah, I agree, not sure there's a need for this test - the more important thing to check would be that the form didn't submit if those conditions were activated but their conditionally-required fields _weren't_ filled out. I.e., your third point. This is already tested on L67.

**Review by peggygannon — APPROVED** [review]
@markgreenburg Content looks good. My only change is that the header should be:  
Apply for disability compensation

**Pull request merged**